core/src/lib/res.cc

```diff
@@ -1337,11 +1338,45 @@ void ConfigurationParser::StoreAddressesPort(LEX* lc,
         || token == BCT_IDENTIFIER)) {
     scan_err1(lc, _("Expected a port number or string, got: %s"), lc->str);
   }
-  if (pass == 1
-      && !AddAddress(GetItemVariablePointer<dlist<IPADDR>**>(*item),
-                     IPADDR::R_SINGLE_PORT, htons(port), AF_INET, 0, lc->str,
-                     errmsg, sizeof(errmsg))) {
-    scan_err2(lc, _("can't add port (%s) to (%s)"), lc->str, errmsg);
+
```

Comment:
in the director's config-parser there is [`DirPort`, `DirAddress` and `DirAddresses`](https://github.com/bareos/bareos/blob/master/core/src/dird/dird_conf.cc#L121-L123) with types `CFG_TYPE_ADDRESSES_PORT`, `CFG_TYPE_ADDRESSES_ADDRESS` and `CFG_TYPE_ADDRESSES_ADDRESSES` mapping to the handler methods `StoreAddressesPort()`, `StoreAddressesAddress()` and `StoreAddressesAddresses()`.

As your change is made to `StoreAddressesPort()`, it will happen when setting `DirPort`, but not `DirAddress`. I don't think it is correct this way.